AdvLoggerPkg/Library/AdvancedLoggerLib/PeiCore/GoogleTest/AdvancedLoggerPeiCoreLib.cpp

```diff
@@ -0,0 +1,173 @@
+/** @file AdvancedLoggerPeiCoreLib.cpp
+
+    This file contains the unit tests for the Advanced Logger PEI Core Library.
+
+    Copyright (c) Microsoft Corporation.
+    SPDX-License-Identifier: BSD-2-Clause-Patent
+**/
+
+#include <Library/GoogleTestLib.h>
+#include <Library/FunctionMockLib.h>
+#include <GoogleTest/Library/MockPeiServicesLib.h>
+#include <GoogleTest/Library/MockMemoryAllocationLib.h>
+#include <GoogleTest/Library/MockHobLib.h>
+#include <GoogleTest/Library/MockAdvancedLoggerHdwPortLib.h>
+#include <GoogleTest/Ppi/MockAdvancedLogger.h>
+
+extern "C" {
+  #include <Uefi.h>
+  #include <Library/BaseLib.h>
+  #include <Library/DebugLib.h>
+  #include <AdvancedLoggerInternal.h>
+  #include <Protocol/AdvancedLogger.h>
+  #include <Protocol/VariablePolicy.h>          // to mock (MU_BASECORE MdeModulePkg)
+  #include <AdvancedLoggerInternalProtocol.h>
+  #include <Library/BaseMemoryLib.h>            // to mock (MU_BASECORE MdePkg)
+  #include <Library/PcdLib.h>                   // to mock OR  NULL lib? (MU_BASECORE MdePkg)
+  #include <Library/SynchronizationLib.h>       // to mock (MU_BASECORE MdePkg)
+  #include <Library/TimerLib.h>                 // to mock (MU_BASECORE MdePkg)
+  #include <Library/VariablePolicyHelperLib.h>  // to mock (MU_BASECORE MdeModulePkg)
+  #include "../../AdvancedLoggerCommon.h"
+
+  #include <Core/Pei/PeiMain.h>
+  #include <Library/MmUnblockMemoryLib.h>
+  #include <Library/PeiServicesTablePointerLib.h> // to mock (MU_BASECORE MdePkg)
+  #include <Library/PrintLib.h>
+}
+
+using namespace testing;
+
+/**
+  Test class for AdvancedLoggerPeiCore
+**/
+class AdvancedLoggerPeiCoreTest : public Test {
+protected:
+  UINTN DebugLevel;
+  CHAR8 *Buffer;
+  UINTN NumberOfBytes;
+  EFI_HANDLE ImageHandle;
+  EFI_SYSTEM_TABLE SystemTable;
+  BOOLEAN status;
+  ADVANCED_LOGGER_INFO testLoggerInfo;
+  // StrictMock<MockHobLib> gHobLib;
+  // StrictMock<MockAdvancedLoggerHdwPortLib> gALHdwPortLib;
+
+  void
+  SetUp (
+    ) override
+  {
+    CHAR8  OutputBuf[] = "MyUnitTestLog";

```

Comment:
This is a local variable, allocated on the stack, that you are assigning to the Buffer.

You will need to either make it a global, or allocate space for it.